Sources/TuistSupportTesting/Utils/MockFileHandling.swift

```diff
@@ -0,0 +1,78 @@
+import Foundation
+@testable import TuistSupport
+import Basic
+
+/// Mock FileHandling without subclassing
+public final class MockFileHandling: FileHandling {
```

Comment:
We can get rid of this one and use the existing mock file handler. If a test subject does IO, you can prepare the test scenario beforehand by writing into disk the files that the subject expects.

Reply:
I'm not sold on this point, I think it's preferable to have a real mock instead of subclassing the `FileHandler` and bringing in unexpected behavior. I also find the real mock easier to use in the tests and more consistent with the testing style

Reply:
> I also find the real mock easier to use in the tests and more consistent with the testing style

I'd not be consistent with how we've been testing subjects that do IO. The theory says that this might introduce flakiness and make tests slower, but we haven't had of the first, and the later is barely perceptible. What we get by doing IO is a test that replicates better what will happen when we use the subject.

Let me give you an example. In your subject, you create a file using file handler. Then in your test using this mock you say: *I expect someone to call the create file method*. That's correct, right? But what if the directory where the file will be created doesn't exist? You won't catch that with your test. That's something that we'd catch if we do real IO because the create file call would error.

Since we don't want to pollute the environment, the file handler that we use for testing purposes points to a temporary directory that is deleted right after each test. That's why there's no flakiness and the behavior of the tests is fully deterministic.